cmake/modules/Nuget.cmake

```diff
@@ -0,0 +1,233 @@
+            message(STATUS "'${nuget_exe_path}' install '${ARG_ID}' -Version '${ARG_VERSION}' -Source '${ARG_SOURCE}' -OutputDirectory '${ARG_OUTPUT_DIR}' -DependencyVersion Ignore -Verbosity quiet")
+            execute_process(
+                COMMAND 
+                ${nuget_exe_path} 
+                install ${ARG_ID}
+                -Version ${ARG_VERSION}
+                -Source ${ARG_SOURCE}
+                -OutputDirectory ${ARG_OUTPUT_DIR}
+                -DependencyVersion Ignore
+                -Verbosity quiet
+                RESULT_VARIABLE result
+            )
+            if(NOT ${result} STREQUAL "0")
+                math(EXPR retry_count "${retry_count} + 1")
+
+                message(STATUS "Nuget failed: '${result}'. Retrying in ${retry_delay} seconds...")
+                execute_process(
+                    COMMAND 
+                    ${CMAKE_COMMAND} -E sleep ${retry_delay}
+                )
+            endif()
+        endwhile()
+
+        if(NOT ${result} STREQUAL "0")
+            message(FATAL_ERROR "NuGet failed: '${result}' Package  '${ARG_NAME}' (${ARG_ID}.${ARG_VERSION})")
+        endif()
+    endif()
+
+    # Set output variable. The NAME parameter is optional: if it's not set then the package ID is used as the
+    # name. The reason for having a separate NAME is for packages whose IDs change (e.g. GDK) so that callers
+    # can use a fixed name in dependents. Example, targets can reference gdk_SOURCE_DIR with the snippet below
+    # instead of having to reference Microsoft.GDK.PC.230300_SOURCE_DIR.
+    #
+    # FetchNuGetPackage(
+    #     NAME gdk
+    #     ID Microsoft.GDK.PC.220300
+    #     VERSION 10.0.22621.3049
+    # )
+    set(${ARG_NAME}_SOURCE_DIR ${ARG_OUTPUT_DIR}/${ARG_ID}.${ARG_VERSION} PARENT_SCOPE)
+endfunction()
+
+# Begin the 'main' logic of this file. Previous code is all defintions.
+message("USE_WARP_FROM_NUGET: ${USE_WARP_FROM_NUGET}")
+if(${USE_WARP_FROM_NUGET} STREQUAL "LATEST_RELEASE" OR ${USE_WARP_FROM_NUGET} STREQUAL "LATEST_PREVIEW")
+
+  message("Fetching warp from nuget")
+
+  FetchNuGetPackage(ID Microsoft.Direct3D.WARP OUTPUT_DIR ${BINARY_DIR}/temp RELEASE_TYPE ${USE_WARP_FROM_NUGET})
+
+  if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
+    set(ARCH "x64")
+  endif()
+  if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "X86")
+    set(ARCH "win32")
+  endif()
+  if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "ARM64")
+    set(ARCH "arm64")
+  endif()
+
+  set(WARP_SOURCE_PATH "${Microsoft.Direct3D.WARP_SOURCE_DIR}/build/native/bin/${ARCH}")
+  set(WARP_DEST_PATH "${BINARY_DIR}/${BUILD_TYPE}/bin/")
+  message("Copying d3d10warp.dll and d3d10warp.pdb \n"
+           "  from:  ${WARP_SOURCE_PATH}\n"
+           "  to: ${WARP_DEST_PATH}")
+  file(COPY "${WARP_SOURCE_PATH}/d3d10warp.dll" 
+       DESTINATION "${WARP_DEST_PATH}")
+  file(COPY "${WARP_SOURCE_PATH}/d3d10warp.pdb" 
+       DESTINATION "${WARP_DEST_PATH}")
+endif()
```

Comment:
Please add an empty line here and also in tools/clang/unittests/HLSLExec/CMakeLists.txt below. Depending on which editor you use there might be a setting to do that automatically.